src/Tokenizer/Tokens.php

```diff
@@ -1284,6 +1233,73 @@ public function clearTokenAndMergeSurroundingWhitespace($index)
         $this->clearAt($nextIndex);
     }
 
+    /**
+     * @param int  $type        type of block, one of BLOCK_TYPE_*
+     * @param int  $searchIndex index of opening or closing brace
+     * @param bool $findEnd     find end or start?
+     *
+     * @return int index of the opposite brace
+     */
+    private function findOppositeBlockEdge($type, $searchIndex, $findEnd)
```

Comment:
for sake of maintability of LTS, this shall be targetting 2.2 branch.
so `findBlockEnd` would be just a proxy method on 2.2, without raising a deprecation

Reply:
btw, if some of the deprecation fixers can be ported to 2.2 line, please do so as well

Reply:
I'm going to open a separate PR to port this to 2.2, but keep this one open for the deprecation and changes to > 2.2 code